graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DgsSchemaProvider.kt

```diff
@@ -508,10 +518,57 @@ class DgsSchemaProvider(
                         enableInstrumentation
 
                     entityFetcherRegistry.entityFetchers[dgsEntityFetcherAnnotation.name] = dgsComponent to method
+
+                    val type = registry.getType(dgsEntityFetcherAnnotation.name)
+
+                    type.ifPresent {
+                        val typeDefinition = it as? ObjectTypeDefinition
```

Comment:
If the type that has a key is an interface, this doesn't work. e.g Movie is an interface with key movieId